unyt/unit_object.py

```diff
@@ -480,18 +480,16 @@ def __pow__(self, p):
 
     def __eq__(self, u):
         """Test unit equality."""
-        if not isinstance(u, Unit):
-            return False
         return (
-            math.isclose(self.base_value, u.base_value)
+            isinstance(u, Unit)
             and self.dimensions == u.dimensions
+            and math.isclose(self.base_value, u.base_value)
+            and math.isclose(self.base_offset, u.base_offset)
```

Comment:
`base_offset` is only nonzero for temperature units. I wonder if it's slightly faster to check if base_offset is non for both and only then calling `math.isclose`.

Reply:
is it ever negative ?

Reply:
Yes, e.g. [`degC`](https://github.com/yt-project/unyt/blob/79ef518e4cb881352ab4395eec0ea69be94b08df/unyt/_unit_lookup_table.py#L131)

Reply:
I had it the other way around. I guess I should have asked if it can be (strictly) positive.

Reply:
Unfortunately it’s public API (via `define_unit`) so I’d rather not change it.

Reply:
I'm not proposing to change it, just naively looking for fast checks that could be used here, but checking for "non-zero" is probably faster anyway.